File: apollo/src/main/java/io/muun/apollo/domain/errors/MessageFromExpiredPairingError.java
package io.muun.apollo.domain.errors;

import io.muun.common.api.messages.MessageSpec;

public class MessageFromExpiredPairingError extends RuntimeException {

    /**
     * Rich constructor.
     */
    public MessageFromExpiredPairingError(String sessionUuid, long messageId, MessageSpec spec) {
        super(String.format(
                "Sender %s sent message %s of type %s, but the pairing was expired",
                sessionUuid,
                Long.toString(messageId),
                spec.messageType
        ));
    }
}

File: apollo/src/main/java/io/muun/apollo/domain/errors/MessageFromExpiredPairingError.kt
package io.muun.apollo.domain.errors

import io.muun.common.api.messages.MessageSpec

class MessageFromExpiredPairingError(sessionUuid: String, messageId: Long, spec: MessageSpec):
    MuunError("Received a message from an expired pairing") {

    init {
        metadata["sessionUuid"] = sessionUuid
        metadata["messageId"] = messageId
        metadata["spec"] = spec.messageType
    }
}